pkg/controller/volume/attachdetach/util/util.go

```diff
@@ -249,3 +276,101 @@ func ProcessPodVolumes(pod *v1.Pod, addVolumes bool, desiredStateOfWorld cache.D
 	}
 	return
 }
+
+func isCSITranslationNeeded(spec *volume.Spec, nodeName types.NodeName, vpm *volume.VolumePluginMgr, csiMigratedPluginManager volume.CSIMigratedPluginManager) (bool, error) {
+	migratable, err := csiMigratedPluginManager.IsMigratable(spec)
+	if err != nil {
+		return false, err
+	}
+	migrationSupportedOnNode, err := isCSIMigrationSupportedOnNode(nodeName, spec, vpm, csiMigratedPluginManager)
+	if err != nil {
+		return false, err
+	}
+	return migratable && migrationSupportedOnNode, err
+}
+
+func isCSIMigrationSupportedOnNode(nodeName types.NodeName, spec *volume.Spec, vpm *volume.VolumePluginMgr, csiMigratedPluginManager volume.CSIMigratedPluginManager) (bool, error) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.CSIMigration) &&
```

Comment:
this function could probably use some more detailed commenting :)

Some context on *why* it's true when these two feature flags is on would be helpful.